client/my-sites/themes/single-site.jsx

```diff
@@ -37,30 +37,31 @@ const ThemesSingleSite = ( props ) => {
 		{ analyticsPath, analyticsPageTitle, isJetpack, translate } = props,
 		jetpackEnabled = config.isEnabled( 'manage/themes-jetpack' );
 
-	if ( isJetpack && jetpackEnabled && ! site.hasJetpackThemes ) {
-		return <JetpackUpgradeMessage site={ site } />;
-	}
-
-	if ( isJetpack && jetpackEnabled && ! site.canManage() ) {
-		return <JetpackManageDisabledMessage site={ site } />;
-	}
-
-	if ( isJetpack && ! jetpackEnabled ) {
-		return (
-			<Main className="themes">
-				<PageViewTracker path={ analyticsPath } title={ analyticsPageTitle }/>
-				<SidebarNavigation />
-				<CurrentTheme
-					site={ site }
-					canCustomize={ site && site.isCustomizable() } />
-				<EmptyContent title={ translate( 'Changing Themes?' ) }
-					line={ translate( 'Use your site theme browser to manage themes.' ) }
-					action={ translate( 'Open Site Theme Browser' ) }
-					actionURL={ site.options.admin_url + 'themes.php' }
-					actionTarget="_blank"
-					illustration="/calypso/images/drake/drake-jetpack.svg" />
-			</Main>
-		);
+	if ( isJetpack ) {
```

Comment:
Huh right, thanks for flattening those conditionals, should've thought of that myself.

Part of this has now gone into 9b89e9d. I haven't yet added a separate method for this since this is currently a stateless functional component -- tell me if you think it reads well enough now or if I should turn it into a normal component :-)